test/performance/observed_concurency_test.go

```diff
@@ -64,28 +60,27 @@ func generateTraffic(client *spoof.SpoofingClient, url string, concurrency int,
 				case <-done:
 					return nil
 				default:
-					res, _ := client.Do(req)
+					res, err := client.Do(req)
+					if err != nil {
+						logger.Infof("Error sending request: %v", err)
+					}
 					if resChannel != nil {
 						resChannel <- res
 					}
-					atomic.AddInt32(&requests, 1)
 				}
 			}
 		})
 	}
 
-	err := group.Wait()
-	requestsMade := atomic.LoadInt32(&requests)
-
-	if err != nil {
-		return requestsMade, fmt.Errorf("error making requests for scale up: %v", err)
+	if err := group.Wait(); err != nil {
+		return fmt.Errorf("error making requests for scale up: %v", err)
```

Comment:
This will never hit today, we only `return nil`.  Should the `resChannel` be replaced with a return?

Reply:
As in the answer above to Markus, I can't really do that since this is executed in a go-routine that also consumes the channel, so it should be precreated.

Reply:
Scratch it, it might return an error if the request is not composed (never saw that happen, but it can in theory).